synthesizer/src/process/stack/inclusion/wasm.rs

```diff
@@ -0,0 +1,192 @@
+// Copyright (C) 2019-2023 Aleo Systems Inc.
+// This file is part of the snarkVM library.
+
+// The snarkVM library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkVM library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkVM library. If not, see <https://www.gnu.org/licenses/>.
+
+use super::*;
+
+impl<N: Network> Inclusion<N> {
+    /// Returns the inclusion assignments for the given execution.
+    pub async fn prepare_execution_wasm(
```

Comment:
If we would like enforce getting inclusion proof data from the web, we'll need an async version of these functions to allow the event loop to be able to create HTTP requests, this should be refactored to avoid code duplication, but this'll be done before the PR is marked green

Reply:
The asyncify & awaitify idea would be like so, but it also seems like trying too hard.

If we do go for making more things async for the browser though it might be a path we want to take

```rust
/// Create both a synchronous and asynchronous version of a function based on a feature flag
#[macro_export]
macro_rules! asyncify {
    ($feature:expr, $($function:tt)*) => {
        #[cfg(feature = $feature)]
        async $($function)*

        #[cfg(not(feature = $feature))]
        $($function)*
    };
}

/// Await or don't await an expression based on a feature flag
#[macro_export]
macro_rules! awaitify {
    ($feature:expr, $expr:expr) => {
        #[cfg(feature = $feature)]
        {
            $expr.await
        }

        #[cfg(not(feature = $feature))]
        {
            $expr
        }
    };
}
```